workflow-service/src/test/java/com/redhat/parodos/ControllerMockClient.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.parodos;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+
+import java.util.Base64;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+public class ControllerMockClient {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Getter
+	@Setter
+	private String validUser = "test";
+
+	@Getter
+	@Setter
+	private String validPassword = "test";
+
+	public MockHttpServletRequestBuilder getRequestWithValidCredentials(String path) {
+		return get(path).header("Authorization", String.format("Basic %s", this.credentials())).header("Content-Type",
+				"application/json");
+	}
+
+	public MockHttpServletRequestBuilder postRequestWithValidCredentials(String path) {
+		return post(path).header("Authorization", String.format("Basic %s", this.credentials())).header("Content-Type",
+				"application/json");
+	}
+
+	public MockHttpServletRequestBuilder getRequestWithInValidCredentials(String path) {
+		return get(path).header("Authorization", "Basic Zm9vOmJhcgo=").header("Content-Type", "application/json");
+	}
+
+	public MockHttpServletRequestBuilder postRequestWithInValidCredentials(String path) {
+		return post(path).header("Authorization", "Basic Zm9vOmJhcgo=").header("Content-Type", "application/json");
```

Comment:
Since the headers are identical, do you think it worth to create a private method to avoid code duplication? In this way, we can also quickly understand that they are identical.